File: GuessTheNumber.js
function guessTheNumber() {
    const secretNumber = Math.floor(Math.random() * 100) + 1; // Generates a random number between 1 and 100

    console.log("Welcome to Guess the Number Game!");

    while (true) {
        const guess = parseInt(prompt("Enter your guess:"));

        if (guess < secretNumber) {
            console.log("Too low! Try again.");
        } else if (guess > secretNumber) {
            console.log("Too high! Try again.");
        } else {
            console.log("Congratulations! You guessed the number.");
            break;
        }
    }
}

guessTheNumber();

File: GuessTheNumber.java
import java.util.Scanner;

public class GuessTheNumber {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int secretNumber = (int) (Math.random() * 100) + 1; // Generates a random number between 1 and 100
        int guess;

        System.out.println("Welcome to Guess the Number Game!");

        do {
            System.out.print("Enter your guess: ");
            guess = scanner.nextInt();

            if (guess < secretNumber) {
                System.out.println("Too low! Try again.");
            } else if (guess > secretNumber) {
                System.out.println("Too high! Try again.");
            } else {
                System.out.println("Congratulations! You guessed the number.");
            }
        } while (guess != secretNumber);

        scanner.close();
    }
}